bucket/qrencode.json

```diff
@@ -0,0 +1,69 @@
+{
+    "version": "4.1.1-3",
+    "description": "Encode input data in a QR Code symbol",
+    "homepage": "https://fukuchi.org/works/qrencode/",
+    "license": "LGPL-2.1-only",
+    "notes": "PNG output is disabled for this build. See https://github.com/msys2/MSYS2-packages/issues/2214.",
+    "architecture": {
+        "64bit": {
+            "url": [
+                "https://mirror.msys2.org/msys/x86_64/libqrencode-4.1.1-3-x86_64.pkg.tar.zst",
+                "https://mirror.msys2.org/msys/x86_64/msys2-runtime-3.3.5-3-x86_64.pkg.tar.zst"
+            ],
+            "hash": [
+                "ead222db63fadf4c6ec67f6b7a95761eb9d0c4ffd92e18129ab11f7944e1e583",
+                "9e99486501905294eb01d7024d2cb96f658b609232878a990e469b5448ea4e11"
+            ]
+        }
+    },
+    "pre_install": [
+        "$keep_files = @(\"$dir\\usr\\bin\\qrencode.exe\",",
+        "                \"$dir\\usr\\bin\\msys-qrencode-4.dll\",",
+        "                \"$dir\\usr\\bin\\msys-2.0.dll\",",
+        "                \"$dir\\usr\\share\\licenses\\libqrencode\\COPYING\",",
+        "                \"$dir\\usr\\share\\man\\man1\\qrencode.1.gz\")",
+        "",
+        "Get-ChildItem \"$dir\\usr\" -Recurse",
+        "    | Select-Object -ExpandProperty FullName",
+        "    | Where-Object {$keep_files -notcontains $_ }",
+        "    | Sort-Object Length -Descending",
+        "    | ForEach-Object { if ((Test-Path $_ -Type Container) -and (Get-ChildItem $_)) { return } Remove-Item $_ }"
```

Comment:
There are three residual files remaining after this (in root folder).

I think this can be simplified by using `Get-ChildItem "$dir" -Recurse -Exclude $keep_files | Remove-Item -Force`

Reply:
> I think this can be simplified by using `Get-ChildItem "$dir" -Recurse -Exclude $keep_files | Remove-Item -Force`

It doesn't work when you try to specify files in folder and subfolders. `-Force` only removes hidden and read-only files. It doesn't stop the prompt and would require the `-Recurse` parameter. `Sort-Object Length -Descending` ensures that files are deleted first before folders are emptied out and can be deleted.

Reply:
Ah, I see